File: #1695.maximum-erasure-value.cpp
class Solution {
public:
    int maximumUniqueSubarray(vector<int>& nums) {

        int size = nums.size();
        int sum = 0;
        int left = 0;
        unordered_set<int> visited;
        int result;
        for (int i = 0; i < size; i++) {
            sum += nums[i]; 
            
            while (visited.count(nums[i])) {
                sum -= nums[left]; visited.erase(nums[left]); 
                left++;
            }
            
            visited.insert(nums[i]);
            result = std::max(result, sum);
        }
        return result;
    }
};

File: #1695.maximum-erasure-value.ts
function maximumUniqueSubarray(nums: number[]): number {
    const set = new Set<number>();
    let maxSum = -Infinity;
    let sum = 0;
    for (let l = 0, r = 0; r < nums.length; r++) {
        while (set.has(nums[r])) {
            set.delete(nums[l]);
            sum -= nums[l];
            l++;
        }
        set.add(nums[r]);
        sum += nums[r]
        maxSum = Math.max(sum, maxSum);
    }
    return maxSum;
};
